src/ptests/test_unstruc_quad_f90.F90

```diff
@@ -0,0 +1,362 @@
+!
+! @file test_unstruc_quad_f90.F90
+! @author Greg Sjaardema <gsjaardema@gmail.com>
+!
+! @section LICENSE
+! BSD style license
+!
+! @section DESCRIPTION
+! Test program for pcgns library
+! -- Created to test cgp_parent_data_write function
+! -- Based on test_unstructured_quad.c
+!
+
+!
+!
+! 2....4....6....8 8...10...12...14 14...16...18...20 L+1.L+3...L+5...L+7
+! |    |    |    | |    |    |    | |     |    |    | |     |     |     |
+! 1....3....5....7 7....9...11...13 13...15...17...19 L...L+2...L+4...L+6
+!   1    2   3      4    5    6       7    8    9      M     M+1   M+2
+!       P0               P1                P2             PN
+!
+!L = P*6+1   M = P*3+1
+!The BC "Bottom" is on the bottom (y=0) edge of the mesh.
+!The BC "Left"   is on the left (x=0) edge of the mesh.
+! -- included to test whether works with 0 entries on some procs.
+!
+
+
+PROGRAM test_unstruc_quad_f
+
+  USE mpi
+  USE ISO_C_BINDING
+  USE CGNS
+  IMPLICIT NONE
+
+#include "cgnstypes_f03.h"
+#ifdef WINNT
+  INCLUDE 'cgnswin_f.h'
+#endif
+  integer, parameter :: dp = kind(0.0d0)
+
+  INTEGER vert_proc
+  PARAMETER(vert_proc = 8)
+  INTEGER depth
+  PARAMETER(depth = 4)
+
+  INTEGER(C_INT) comm_size, comm_rank, mpi_err
+  INTEGER info
+  INTEGER ierr
+  INTEGER(cgsize_t) :: sizes(3)
+  INTEGER(cgsize_t) :: n_boco_elems
+  INTEGER(cgsize_t) :: start_local(1), end_local(1)
+  INTEGER(cgsize_t) :: start, end
+  INTEGER(cgsize_t) :: emin(1), emax(1)
+  INTEGER(cgsize_t) :: rmin(1), rmax(1)
+  INTEGER(cgsize_t) :: nelem, nvert
+  INTEGER(cgsize_t), PARAMETER :: start_1 = 1
+  INTEGER Cx, Cy, Cz, F, B, Z, S, BC
+  INTEGER i, k
+  INTEGER global_num_quads
+  INTEGER(C_INT), DIMENSION(4) :: indices
+
+  REAL(dp), allocatable ::  fx(:), fy(:), fz(:) 
+  INTEGER(cgsize_t), allocatable, target :: elements(:)
+  INTEGER(cgsize_t), allocatable :: point_list(:)
+  integer(cgsize_t), dimension(:), pointer :: el_ptr => null()
+  integer(cgsize_t), dimension(:), pointer :: null_ptr => null()
+  LOGICAL found_point
+
+
+  CHARACTER(len=10, kind=C_CHAR), allocatable, target :: labels(:)
+  TYPE(C_PTR), dimension(:), allocatable :: pt_labels
+! First define the label array on fortran side 
+  allocate(labels(4))
+  labels(1) = "Zone_t"//C_NULL_CHAR
+  labels(2) = "ZoneBC_t"//C_NULL_CHAR
+  labels(3) = "BC_t"//C_NULL_CHAR
+  labels(4) = "PointList"//C_NULL_CHAR
+! do mapping for C side
+  allocate(pt_labels(4))
+  pt_labels(1) = C_LOC(labels(1))
+  pt_labels(2) = C_LOC(labels(2))
+  pt_labels(3) = C_LOC(labels(3))
+  pt_labels(4) = C_LOC(labels(4))
+
+!
+!---- initialize MPI
+  CALL MPI_INIT(mpi_err)
```

Comment:
You should move the MPI_INIT before any executable lines

Reply:
done